File: 0071-simplify-path/0071-simplify-path.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
    answer = []
    splitedpath = path.split("/")
    for(p of splitedpath){
        if(p == '' || p == "."){
            continue;
        }else if(p == ".."){
            if(answer.length > 0){
                answer.pop();
            }
        }else{
            answer.push(p);
        }
    }
    return "/" + answer.join('/')
}

File: leetcode/Medium/0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        answer = []
        l = [x for x in path.split("/") if not x == ""]
        for i in l:
            if i == ".":
                continue
            elif i == "..":
                if len(answer) > 0:
                    answer.pop()
            else:
                answer.append(i)
        return "/" + "/".join(answer)
